projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+    UserResponse
+} from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken, DspApiConnectionToken } from './core.module';
+
+/**
+ * Currently logged-in user information
+ */
+export interface CurrentUser {
+    // username
+    name: string;
+
+    // json web token
+    jwt: string;
+
+    // default language for ui
+    lang: string;
+
+    // is system admin?
+    sysAdmin: boolean;
+
+    // list of project shortcodes where the user is project admin
+    projectAdmin: string[];
+}
+
+/**
+ * Session with id (= login timestamp) and inforamtion about logged-in user
+ */
+export interface Session {
+    id: number;
+    user: CurrentUser;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SessionService {
+
+    /**
+     * max session time in milliseconds
+     * default value (24h): 86400000
+     *
+     */
+    readonly MAX_SESSION_TIME: number = 86400000; // 1d = 24 * 60 * 60 * 1000
+
+
+    constructor(
+        @Inject(DspApiConnectionToken) private dspApiConnection: KnoraApiConnection,
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig
+    ) { }
+
+    /**
+     * get session information from localstorage
+     */
+    getSession(): Session {
+        if (localStorage.getItem('session') !== null) {
+            return JSON.parse(localStorage.getItem('session'));
+        }
+
+        return undefined;
+    }
+
+    /**
+     * set session by using the json web token (jwt) and the user object;
+     * it will be used in the login process
+     *
+     * @param jwt
+     * @param username
+     */
+    setSession(jwt: string, identifier: string, identifierType: 'email' | 'username') {
+
+        let session: Session;
+
+        if (jwt) {
+            this.updateDspApiConnection(jwt);
+        }
+
+        // get user information
+        this.dspApiConnection.admin.usersEndpoint.getUser(identifierType, identifier).subscribe(
+            (response: ApiResponseData<UserResponse>) => {
+                let sysAdmin = false;
+                const projectAdmin: string[] = [];
+
+                // get permission information: a) is user sysadmin? b) get list of project iri's where user is project admin
+                const groupsPerProjectKeys: string[] = Object.keys(response.body.user.permissions.groupsPerProject);
+
+                for (const key of groupsPerProjectKeys) {
+                    if (key === Constants.SystemProjectIRI) {
+                        sysAdmin = response.body.user.permissions.groupsPerProject[key].indexOf(Constants.SystemAdminGroupIRI) > -1;
+                    }
+
+                    if (response.body.user.permissions.groupsPerProject[key].indexOf(Constants.ProjectAdminGroupIRI) > -1) {
+                        projectAdmin.push(key);
+                    }
+                }
+
+                // store session information in browser's localstorage
+                // TODO: jwt will be removed, when we have a better cookie solution (DSP-261)
+                session = {
```

Comment:
Make a constructor for `CurrentUser` and `Session` and instantiate the class using `new`.

Reply:
Or maybe just make it an interface since you have to deserialize it from json. It would be good to add a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types). TypeScript has a structural type system anyway. So you could juts check if the object has the necessary properties after deserialisation.